src/encoder.rs

```diff
@@ -2289,13 +2288,13 @@ mod tests {
 /// Since this only contains trait impls, there is no need to make this public, they are simply
 /// available when the mod is compiled as well.
 impl Compression {
-    fn to_options(self) -> deflate::CompressionOptions {
+    fn to_options(self) -> flate2::Compression {
         match self {
-            Compression::Default => deflate::CompressionOptions::default(),
-            Compression::Fast => deflate::CompressionOptions::fast(),
-            Compression::Best => deflate::CompressionOptions::high(),
-            Compression::Huffman => deflate::CompressionOptions::huffman_only(),
-            Compression::Rle => deflate::CompressionOptions::rle(),
+            Compression::Default => flate2::Compression::default(),
+            Compression::Fast => flate2::Compression::fast(),
+            Compression::Best => flate2::Compression::best(),
+            Compression::Huffman => flate2::Compression::fast(), // TODO
+            Compression::Rle => flate2::Compression::fast(),     // TODO
```

Comment:
Could you implement these two as `flate2::Compression::none()` instead? That way they'd still technically uphold the same guarantees as before

Reply:
Done. Also bumped crate version so I could reference the version in the deprecation notice.